compiler/rustc_data_structures/src/stable_hasher.rs

```diff
@@ -137,7 +137,35 @@ impl Hasher for StableHasher {
         // platforms. This is important for symbol hashes when cross compiling,
         // for example. Sign extending here is preferable as it means that the
         // same negative number hashes the same on both 32 and 64 bit platforms.
-        self.state.write_i64((i as i64).to_le());
+        let value = (i as i64).to_le() as u64;
+
+        // Cold path
+        #[cold]
+        #[inline(never)]
```

Comment:
Afaik `[cold]` is implemented as `coldcc`, which already prevents inlining. So the 2nd annotation should be redundant.
https://llvm.org/docs/LangRef.html#calling-conventions

Reply:
I searched through `rustc` sources and most uses of `#[cold]` actually also add `#[inline(never)]` :grin: But I agree, it's most probably redundant, I'll remove it if perf is good.

Reply:
Btw @the8472 it seems that `#[inline(never)]` is required :thinking: https://godbolt.org/z/qrsh3WahK If you uncomment `#[inline(never)]`, it won't be inlined, if you comment it, it will be inlined.